### PR TITLE
Fix GetValidators

### PR DESCRIPTION
**Pull request by Tommo-L**
Prevent `StandbyValidators` from being modified

### CODE REVIEW

File: tests/neo.UnitTests/SmartContract/Native/Tokens/UT_NeoToken.cs [hunk 1]
```diff
@@ -389,6 +389,8 @@ public void TestGetValidators2()
         {
             var snapshot = Blockchain.Singleton.GetSnapshot();
             var result = NativeContract.NEO.GetValidators(snapshot);
+            result[0] = result[1]; // tampering the data
+            result = NativeContract.NEO.GetValidators(snapshot);
```
Comment (vncoelho) [inline tests/neo.UnitTests/SmartContract/Native/Tokens/UT_NeoToken.cs]: The UT could ensure that the references are different, right?

Comment (Tommo-L) [reply tests/neo.UnitTests/SmartContract/Native/Tokens/UT_NeoToken.cs]: Yes

**Comment by erikzhang** [comment]
What's the difference?

**Comment by vncoelho** [comment]
I believe it is because of the reference being returned instead of a new array.

**Comment by Tommo-L** [comment]
> I believe it is because of the reference being returned instead of a new array.

Yes. If there no voting, `Blockchain.StandbyValidators` will be returned directly, and it can be tampered with outside, like in OracleContract.

```c#
 public ECPoint[] GetOracleValidators(StoreView snapshot)
        {
            ECPoint[] oraclePubKeys = NativeContract.NEO.GetValidators(snapshot);  // <-----
            for (int index = 0; index < oraclePubKeys.Length; index++)
            {
                var oraclePubKey = oraclePubKeys[index];
                StorageKey key = CreateStorageKey(Prefix_Validator, oraclePubKey);
                ECPoint delegatePubKey = snapshot.Storages.TryGet(key)?.Value.AsSerializable<ECPoint>();
                if (delegatePubKey != null) { oraclePubKeys[index] = delegatePubKey; }  // <-----
            }
            return oraclePubKeys;
        }
```

**Comment by erikzhang** [comment]
If it is in a contract, it will be wrapped to a `VMArray`. If it is in a native contract, it is controlled by us. So I can't see any difference.

**Comment by Tommo-L** [comment]
> If it is in a contract, it will be wrapped to a VMArray. If it is in a native contract, it is controlled by us.

That's right. But in this case, shoud we keep the data consistency of `NativeContract.NEO.GetValidators`. Currently, it's a little strange, which we can modify the validators directly in some cases, and in some cases, we can't.  On other hand, if we really want to update the validators, it's recommend to use `setValidators` method or some other ways.

**Pull request closed without merging**